### DISCUSSION TITLE
Can i change view from viewmodel?

### DISCUSSION THREAD

**Discussion by mootfrost** [AvaloniaUI/Avalonia#8294] [Q&A]
I have 2 views. I want to change view from registration to main, but I don't know how to do it.

**Comment by maxkatz6** [comment]
Simplest navigation would be to have an array of children view models - tabs - and bind them to the TabControl, with different DataTemplates per page view model type. 
More complex navigation might require more work, good example was here https://github.com/wieslawsoltes/NavigationSample/tree/master/NavigationSample

It has quite a bit of boilerplate, code required to setup everything. So depending on your MVVM framework for view models, ReactiveUI or Prism or other, you can simplify it. For instance, simple ReactiveUI navigation with screens https://www.reactiveui.net/docs/handbook/routing/

**Reply by maxkatz6** [reply]
Also from the our docs, same reactiveui navigation
https://docs.avaloniaui.net/guides/deep-dives/reactiveui/routing
